exporter/awsemfexporter/metric_translator.go

```diff
@@ -264,17 +265,24 @@ func getGroupedMetrics(metric *pdata.Metric, namespace string, instrumentationLi
 
 // getGroupedMetricKey generates a key for a given GroupedMetric
 func getGroupedMetricKey(cwNamespace string, timestamp int64, labels map[string]string) (string) {
-	var keySlice []string
-	nameSpace := namespaceKey + cwNamespace
-	timeStamp := "Timestamp:" + strconv.FormatInt(timestamp, 10)
-	keySlice = append(keySlice, nameSpace, timeStamp)
+	keySlice := make([]string, 0, len(labels) + 2)
+	fields := make(map[string]string)
+	fields[namespaceKey] = cwNamespace
+	fields[timestampKey] = strconv.FormatInt(timestamp, 10)
+	keySlice = append(keySlice, namespaceKey, timestampKey)
 
 	for k, v := range labels {
-		keyValuePair := k + ":" + v 
-		keySlice = append(keySlice, keyValuePair)
+		fields[k] = v
+		keySlice = append(keySlice, k)
 	}
+
 	sort.Strings(keySlice)
-	key := strings.Join(keySlice, ",")
+	fieldsSlice := make([]string, 0, len(keySlice))
+	for _, v := range keySlice {
+		keyValuePair := v + ":" + fields[v]
+		fieldsSlice = append(fieldsSlice, keyValuePair)
+	}
+	key := strings.Join(fieldsSlice, ",")
```

Comment:
would it be better here to just construct the key string directly instead of creating a slice of kv strings and _then_ concatenating?

Reply:
resolved